torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,185 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    # When is_view is false, the tensor is the base.
+    # When is_view is true, size, stride and storage_offset are all None.
+    is_view: bool = True
+
+    def regenerate_view(self, bases_list: List[Tensor]):
+        if not self.is_view:
+            return bases_list[self.base_index]
+        return torch.as_strided(
+            bases_list[self.base_index],
+            self.size,
+            self.stride,
+            self.storage_offset,
+        )
+
+
+# arg_to_base_index maps arg_name to ViewInfo | [ViewInfo]
+def write_view_information_to_args(
+    arg_names: List[str],
+    arg_types: List[torch.Type],
+    input_kwargs: Dict[str, Any],
+    output_kwargs: Dict[str, Any],
+    arg_to_base_index: Dict[str, Any],
+):
+    def write_single_view(prefix: str, tensor: Tensor, base_index: int):
+        assert f"{prefix}_base_index" not in output_kwargs
+        assert f"{prefix}_size" not in output_kwargs
+        assert f"{prefix}_stride" not in output_kwargs
+        assert f"{prefix}_storage_offset" not in output_kwargs
+
+        if tensor is None:
+            output_kwargs[f"{prefix}_base_index"] = None
+        elif tensor._base is None:
+            # if the tensor is an alias of base (not view), for simplicity we do not serialize view meta.
+            output_kwargs[f"{prefix}_base_index"] = base_index
+            return
+        else:
+            output_kwargs[f"{prefix}_base_index"] = base_index
+            output_kwargs[f"{prefix}_size"] = tensor.size()
+            output_kwargs[f"{prefix}_stride"] = tensor.stride()
+            output_kwargs[f"{prefix}_storage_offset"] = tensor.storage_offset()
+
+    for arg_name, arg_type in zip(arg_names, arg_types):
+        arg = input_kwargs[arg_name]
+        if isinstance(arg_type, torch.ListType):
+            if arg is None:
+                output_kwargs[f"_{arg_name}_length"] = None
+
+            output_kwargs[f"_{arg_name}_length"] = len(arg)
+            for i, elem in enumerate(arg):
+                write_single_view(
+                    f"_{arg_name}_{i}", elem, arg_to_base_index[arg_name][i]
+                )
+
+        elif isinstance(arg_type, (torch.TensorType, torch.OptionalType)):
+            write_single_view(
+                f"_{arg_name}",
+                input_kwargs[arg_name],
+                arg_to_base_index.get(arg_name, None),
+            )
+        else:
+            raise RuntimeError(f"Unsupported type {arg_type}")
+
+
+# Returns a dict of arg_name -> ViewInfo | [ViewInfo]
+def read_view_information_from_args(
+    arg_names: List[str],
+    arg_types: List[torch.Type],
+    input_kwargs: Dict[str, Any],
+    all_bases: List[Tensor],
+    pop_args: bool,
```

Comment:
Do you ever call this with pop_args = False? It doesn't look like it

Reply:
not anymore, since we removed the fancy log message haha